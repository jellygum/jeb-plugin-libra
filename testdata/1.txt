CompiledProgram: {
Modules: [
],
Script: CompiledScript: {
Main:
	public 0x0.<SELF>.main(): ()
		locals(0): Integer,
		LdConst(0)
		StLoc(0)
		LdTrue
		BrFalse(14)
		CopyLoc(0)
		LdConst(5)
		Ge
		BrFalse(9)
		Branch(14)
		MoveLoc(0)
		LdConst(1)
		Add
		StLoc(0)
		Branch(2)
		LdConst(42)
		MoveLoc(0)
		LdConst(5)
		Eq
		Assert
		Ret
Struct Handles: []
Module Handles: [
	0x0.<SELF>,]
Function Handles: [
	0x0.<SELF>.main(): (),]
Type Signatures: []
Function Signatures: [
	(): (),]
Locals Signatures: [
	Integer,]
Strings: [
	<SELF>,
	main,]
ByteArrays: []
Addresses: [
	0x0,]
}

}
