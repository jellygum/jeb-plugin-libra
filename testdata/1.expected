Module Handles: [
    0x0.<SELF>,
]
Struct Handles: [
]
Function Handles: [
    0x0.<SELF>.main(): (),
]
Struct Definitions: [
]
Field Definitions: [
]
Type Signatures: [
]
Function Signatures: [
    (): (),
]
Local Signatures: [
    integer,
]
Strings: [
    <SELF>,
    main,
]
ByteArrays: [
]
Addresses: [
    0x0,
]
Function Definitions: [
]
public 0x0.<SELF>.main(): ()
    max_stack_size: 3
    locals: integer
    LD_CONST ImmUint64(0)              [0,1]
    ST_LOC IdxLocal(0)                 [1,0]
    LD_TRUE                            [0,1]
    BR_FALSE Branch(14)                [1,0]
    COPY_LOC IdxLocal(0)               [0,1]
    LD_CONST ImmUint64(5)              [1,2]
    GE                                 [2,1]
    BR_FALSE Branch(9)                 [1,0]
    BRANCH Branch(14)                  [0,0]
    MOVE_LOC IdxLocal(0)               [0,1]
    LD_CONST ImmUint64(1)              [1,2]
    ADD                                [2,1]
    ST_LOC IdxLocal(0)                 [1,0]
    BRANCH Branch(2)                   [0,0]
    LD_CONST ImmUint64(42)             [0,1]
    MOVE_LOC IdxLocal(0)               [1,2]
    LD_CONST ImmUint64(5)              [2,3]
    EQ                                 [3,2]
    ASSERT                             [2,0]
    RET                                [0,0]

